### ISSUE TITLE
Package uninstall not removing reference

### ISSUE THREAD

**Issue by tgillitzer** [NuGet/Home#426]
I'm trying to create NuGet package of a library which is built by a third party provider.  I am attempting to host this in my internal NuGet repository (ProGet) for inclusion in internal projects (this is our standard way of including libraries rather than putting them directly in source control or some other mechanism).

For some reason unknown to me, NuGet packages created with this library, when the package is uninstalled from a project, the project reference is not removed.

I've beat my head against this wall quite a bit and have looked at all the documentation I can find, but can't seem to find any details about the "magic" around how NuGet adds and removes references for included files.

In troubleshooting this, I have seen different behaviors around the name of the file as it is included in the nupkg file, but I can't come up with anything concrete.

I have messed around with a few different libraries and seen different behaviors, but the original file I was attempting to work on was the FlowSSH library from Bitvise (http://www.bitvise.com/flowssh). 

Any help would be greatly appreciated.

My nuspec file looks like this:
![image](https://cloud.githubusercontent.com/assets/11950389/7147855/a7abe508-e2c3-11e4-9e75-240c605bb6a2.png)

**Comment by deepakaravindr** [comment]
@tgillitzer 
Please provide more details
Which version of NuGet are you using? Is it VS 2015? Which reference is not getting removed? Because, uninstallation of a package does not remove framework references

**Comment by tgillitzer** [comment]
VS 2012 and VS 2013.

2012 NuGet extension version 2.8.50313.31
2013 NuGet extension version 2.7.40911.287

In my example above, the reference to Test is not being removed.

As a side note, (at least partially) confirming the file name theory, it turns out the filename of the Bitvise library I ran into this problem on must be named a certain name (FlowSshNet32.dll).  Once I named it that, the reference is removed as expected.  The filename as provided when downloading FlowSshNet is FlowSshNet527_Framework40_x86.dll.  If in the Nuspec, the target is specified as any name other than FlowSshNet32.dll, the reference is not removed at package uninstall.

It seems like this is probably as designed, however, it would be nice to have some documentation as to what the details of making reference install and uninstall work.

Thanks
